internal/services/storage/storage_account_resource.go

```diff
@@ -1589,6 +1589,15 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 			}
 		}
 
+		if dnsEndpointType == string(storage.DNSEndpointTypeAzureDNSZone) {
+			if blobProperties.RestorePolicy != nil && blobProperties.RestorePolicy.Enabled != nil && *blobProperties.RestorePolicy.Enabled {
+				// Otherwise, API returns: "Required feature Global Dns is disabled"
+				// This is confirmed with the SRP team, where they said:
+				// > restorePolicy feature is incompatible with partitioned DNS
```

Comment:
did they mention if this is a temporary limitation, or something that's planned to be removed in the future?

Reply:
@tombuildsstuff I'll check with them and update it here later..

Reply:
@magodo any update on this?

Reply:
Sorry for the delay (as I was on vacation), I just got the reply from the service team that this is a temporary limitation, and they said they don't have an ETA when this will be supported.

Reply:
No worries - hope you enjoyed the time off :)

Cool ok, in that case can we update this comment to clarify that this is a temporary limitation rather than long-term limitation and so we should re-check this in the future? Perhaps it'd also be worth linking to this thread too?

Reply:
@tombuildsstuff Sure, I've added the comment.